### PR TITLE
feat: add C implementation for `math/base/special/coth`

### LINKED ISSUE #754: [RFC]: Add C implementation for `@stdlib/math/base/special/coth`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/coth`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/coth).

```c
double stdlib_base_coth( const double x );
```

### Related Issues

#752 .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by lovelindhoni**
Resolves #754  .

## Description

This pull request adds native C implementation for `@stdlib/math/base/special/coth`

> What is the purpose of this pull request?

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #754 
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by lovelindhoni** [comment]
Apologies for rewriting history on #1659. It won't happen again.

File: lib/node_modules/@stdlib/math/base/special/coth/LICENSE [hunk 1]
```diff
@@ -0,0 +1,195 @@
+                                Apache License
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/LICENSE]: please revert this

File: lib/node_modules/@stdlib/math/base/special/coth/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2022 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/README.md]:
```suggestion
Copyright (c) 2022 The Stdlib Authors.
```

We won't update copyright year for file change.

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
LGTM. Thanks @lovelindhoni, please apply requested changes and this PR will be ready to merge.

File: lib/node_modules/@stdlib/math/base/special/coth/LICENSE (hunk 1 above)
Comment (lovelindhoni) [reply lib/node_modules/@stdlib/math/base/special/coth/LICENSE]: uh.. I should leave an empty line at the start of the file right?

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/coth/LICENSE]: @lovelindhoni Just remove the file; we don't have a separate Apache LICENSE file in each package directory.

File: lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/Makefile [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#/
+# Removes generated files.
+#
+# @example
+# make clean
+#/
+clean:
+	$(QUIET) -rm -f *.o *.out
+
+.PHONY: clean
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/Makefile]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::native::%s\n", NAME );
+		elapsed = benchmark();
+		print_results( elapsed );
+		printf( "ok %d benchmark finished\n", i+1 );
+	}
+	print_summary( REPEATS, REPEATS );
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/benchmark/c/benchmark.c]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/binding.gyp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+          'action': [
+            'cp',
+            '<(PRODUCT_DIR)/<(addon_target_name).node',
+            '<(addon_output_dir)/<(addon_target_name).node',
+          ],
+        },
+      ], # end actions
+    }, # end target copy_addon
+  ], # end targets
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/binding.gyp]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/examples/c/Makefile [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+#/
+# Removes generated files.
+#
+# @example
+# make clean
+#/
+clean:
+	$(QUIET) -rm -f *.o *.out
+
+.PHONY: clean
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/examples/c/Makefile]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+int main( void ) {
+    const double x[] = { -4.0, -3.11, -2.22, -1.33, -0.44, 0.44, 1.33, 2.22, 3.11, 4.0 };
+
+    double v;
+    int i;
+    for ( i = 0; i < 10; i++ ) {
+        v = stdlib_base_coth( x[ i ] );
+        printf( "coth(%lf) = %lf\n", x[ i ], v );
+    }
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/examples/c/example.c]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/include.gypi [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    'libraries': [
+      '<!@(node -e "var arr = require(\'@stdlib/utils/library-manifest\')(\'./manifest.json\',{},{\'basedir\':process.cwd(),\'paths\':\'posix\'}).libraries; for ( var i = 0; i < arr.length; i++ ) { console.log( arr[ i ] ); }")',
+    ],
+
+    # Library directories:
+    'library_dirs': [
+      '<!@(node -e "var arr = require(\'@stdlib/utils/library-manifest\')(\'./manifest.json\',{},{\'basedir\':process.cwd(),\'paths\':\'posix\'}).libpath; for ( var i = 0; i < arr.length; i++ ) { console.log( arr[ i ] ); }")',
+    ],
+  }, # end variables
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/include.gypi]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/include/stdlib/math/base/special/coth.h [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+/**
+* Computes the hyperbolic cotangent of a double-precision floating-point number.
+*/
+double stdlib_base_coth( const double x );
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif // !STDLIB_MATH_BASE_SPECIAL_COTH_H
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/include/stdlib/math/base/special/coth.h]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/mainfest.json [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        "./include"
+      ],
+      "libraries": [],
+      "libpath": [],
+      "dependencies": [
+       "@stdlib/math/base/special/tanh"
+      ]
+    }
+  ]
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/mainfest.json]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/src/Makefile [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+#/
+# Removes generated files.
+#
+# @example
+# make clean
+#/
+clean: clean-addon
+
+.PHONY: clean
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/src/Makefile]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/src/addon.c [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/coth.h"
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_D_D( stdlib_base_coth )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/src/addon.c]: newline

File: lib/node_modules/@stdlib/math/base/special/coth/src/main.c [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+* @return     output value
+*
+* @example
+* double out = stdlib_base_coth( 0.0 );
+* // returns INFINITY
+*/
+
+double stdlib_base_coth( const double x ) {
+    return 1.0 / stdlib_base_tanh(x);
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/coth/src/main.c]: newline

**Review by Pranavchiku — APPROVED** [review]
Looks good, just add newline to files requested below. This will be good to merge then.

**Review by Planeshifter — APPROVED** [review]
LGTM. Let's get this in; thanks!

**Pull request merged**